src/ir_Mitsubishi.cpp

```diff
@@ -292,6 +340,128 @@ void IRsend::sendMitsubishiAC(unsigned char data[], uint16_t nbytes,
+          && matchSpace(results->rawbuf[offset++], MITSUBISHI_AC_HDR_SPACE);
+    }
+    if (!headerFound) {
+      DPRINTLN("Header mark not found.");
+      failure = true;
+    }
+// Decode byte-by-byte:
+    for (uint8_t i = 0; i < MITSUBISHI_AC_STATE_LENGTH && !failure; i++) {
+      results->state[i] = 0;
+      if (!matchByte(&results->state[i], &(results->rawbuf[offset]),
+          MITSUBISHI_AC_BIT_MARK, MITSUBISHI_AC_ONE_SPACE,
+          MITSUBISHI_AC_ZERO_SPACE)) {
+        DPRINT("Byte decode failed at #");
+        DPRINTLN((uint16_t)i);
+        failure = true;
+      } else {
+        offset+=16;  // 8 marks + 8 spaces
+        DPRINT((uint16_t)results->state[i]);
+        DPRINT(",");
+      }
+      DPRINTLN("");
+    }
+// HEADER validation:
+    if (failure || results->state[0] != 0x23 || results->state[1] != 0xCB ||
+        results->state[2] != 0x26 || results->state[3] != 0x01 ||
+        results->state[4] != 0x00) {
+      DPRINTLN("Header mismatch.");
+      failure = true;
+    } else {
+// DATA part:
+
+// FOOTER checksum:
+      if (IRMitsubishiAC::calculateChecksum(results->state) !=
+          results->state[17]) {
+        DPRINTLN("Checksum error.");
+        failure = true;
+      }
+    }
+    if (rep != MITSUBISHI_AC_MIN_REPEAT && failure) {
+      bool repeatMarkFound = false;
+      while (!repeatMarkFound && offset <
+          (results->rawlen  - (MITSUBISHI_AC_STATE_LENGTH * 8 * 2 + 4))) {
+        repeatMarkFound = matchMark(results->rawbuf[offset++],
+            MITSUBISHI_AC_RPT_MARK) &&
+            matchSpace(results->rawbuf[offset++], MITSUBISHI_AC_RPT_SPACE);
+      }
+      if (!repeatMarkFound) {
+        DPRINTLN("First attempt failure and repeat mark not found.");
+        return false;
+      }
+    }
+    rep++;
+// Check if the repeat is correct if we need strict decode:
+    if (strict && !failure) {
+      DPRINTLN("Strict repeat check enabled.");
+// Repeat mark and space:
+      if (!matchMark(results->rawbuf[offset++], MITSUBISHI_AC_RPT_MARK) ||
+          !matchSpace(results->rawbuf[offset++], MITSUBISHI_AC_RPT_SPACE)) {
+      DPRINTLN("Repeat mark error.");
+      return false;
+    }
+// Header mark and space:
+      if (!matchMark(results->rawbuf[offset++], MITSUBISHI_AC_HDR_MARK) ||
+          !matchSpace(results->rawbuf[offset++], MITSUBISHI_AC_HDR_SPACE)) {
+        DPRINTLN("Repeat header error.");
+        return false;
+      }
+// Payload:
+      uint8_t data;
+      for (uint8_t i = 0; i < MITSUBISHI_AC_STATE_LENGTH; i++) {
+        if (!matchByte(&data, &(results->rawbuf[offset]),
+            MITSUBISHI_AC_BIT_MARK, MITSUBISHI_AC_ONE_SPACE,
+            MITSUBISHI_AC_ZERO_SPACE) || data != results->state[i]) {
+          DPRINTLN("Repeat payload error.");
+          return false;
+        }
+        offset+=16;  // 8 marks + 8 spaces
+      }
+    }  // strict repeat check
+  } while (failure && rep <= MITSUBISHI_AC_MIN_REPEAT);
+  results->decode_type = MITSUBISHI_AC;
```

Comment:
Need to also record the number of bits as per every other decoder.

Reply:
Valid.